server/expressApp.js

```diff
@@ -25,6 +29,24 @@ function newExpressApp (options, middlewares, securityService) {
     securityService.applyAuthMiddleware(app, graphqlEndpoint)
   }
 
+  if (auditLogEnabled) {
+    app.use(function (req, res, next) {
+      try {
+        req.clientInfo = getClientInfoFromHeaders(req)
+      } catch (e) {
+        log.error('Error getting client info.')
+        log.error(e)
+        const message = 'Error decoding/parsing malformed client info. Message: ' + e.message
```

Comment:
[optional] e.message could be exposed only on the development version. 
Generally best to not expose messages from errors. In this case we have control over this so it's not a big deal.

Reply:
{quote}
Any reason why we need to handle this try/catch here (we could use global error handler for express and attach error code there)
{quote}

That ends up in a 500 error, not 400.

Reply:
@aliok yeah - so the common pattern is to supply status code to error object and global error handle reading status code and using it to send the message.
```javascript
const error = new Error("error msg")
error.statusCode = 400
error.payload= {...}
```

Reply:
@wtrocki 
But, in that case, don't we move the HTTP status related business logic into `getClientInfoFromHeaders` which is in `logger`?

Unless I misunderstood, I don't like doing this in `logger` module:
```
const error = new Error("error msg")
error.statusCode = 400
error.payload= {...}
```

Reply:
@aliok We are moving towards talking about opinionated ways of handling errors. I do not have strong opinion so let's leave this untouched. 
Just pointed out that this is usually handled this way as it will allow us to enforce single format of the error payload etc.
Resolving problem above should be enough to get that merged IMHO.

Reply:
@wtrocki could you go a bit more in detail please? perhaps I didn't understand what you meant initially.

Reply:
I think that your example clearly shows what I intended - throw error with all information that generic error handler can use to build.